File: site/js/services/LikeService.js
import { CONFIG, AUTH, API } from "../settings/config.js";
export class LikeService {
    constructor(requestService) {
        this.sendRequest = requestService.sendRequest;
        this.setPostLike = this.setPostLike.bind(this);
        this.getPostLikes = this.getPostLikes.bind(this);
    }
    async setPostLike(accessToken, postId) {
        await this.sendRequest(`${API.setLike}?postId=${postId}`, {
            method: "POST",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
    }
    async getPostLikes(accessToken, postId) {
        const response = await this.sendRequest(`${API.postLikes}?postId=${postId}`, {
            method: "GET",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
        return (await response.json())["COUNT(*)"];
    }
    async getProductLikes(accessToken, productId) {
        const response = await this.sendRequest(`${API.getProductLikes}?productId=${productId}`, {
            method: "GET",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
        return (await response.json())["COUNT(*)"];
    }
    async setProductLike(accessToken, productId) {
        await this.sendRequest(`${API.setProductLike}?productId=${productId}`, {
            method: "POST",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
    }
    async getUserLikedProducts(accessToken) {
        const response = await this.sendRequest(`${API.getUserLikedProducts}`, {
            method: "GET",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
        return await response.json();
    }
    async getLikedPosts({ accessToken, limit = 50, offset = 0 }) {
        const response = await this.sendRequest(`${API.getLikedPosts}?limit=${limit}&offset=${offset}`, {
            method: "GET",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
        return await response.json();
    }
    async getLikedProducts({ accessToken, limit = 50, offset = 0 }) {
        const response = await this.sendRequest(`${API.getLikedProducts}?limit=${limit}&offset=${offset}`, {
            method: "GET",
            headers: { Authorization: `Bearer ${accessToken}` }
        });
        return await response.json();
    }
}

File: server/services/like_service.py
from settings.config import *

class LikeService():
    def __init__(self, like_database):
        self.like_database = like_database

    async def set_post_like(self, post_id, user_id):
        is_user_liked_post = await self.like_database.get_post_user_like(post_id, user_id)

        if is_user_liked_post == None:
            await self.like_database.add_like(post_id, user_id)
        
        else:
            await self.like_database.remove_like(post_id, user_id)
    
    async def get_post_likes(self, post_id):
        return await self.like_database.get_post_likes(post_id)
    
    async def set_product_like(self, product_id, user_id):
        is_user_liked_product = await self.like_database.get_product_user_like(product_id, user_id)

        if is_user_liked_product == None:
            await self.like_database.add_product_like(product_id, user_id)
        
        else:
            await self.like_database.remove_product_like(product_id, user_id)

    async def get_user_liked_products(self, user_id):
        return await self.like_database.get_user_liked_products(user_id)
    
    async def get_liked_posts(self, user_id, limit, offset):
        return await self.like_database.get_liked_posts(user_id, limit, offset)
    
    async def get_liked_products(self, user_id, limit, offset):
        return await self.like_database.get_liked_products(user_id, limit, offset)
    
    async def get_product_likes(self, product_id):
        return await self.like_database.get_product_likes(product_id)